src/main/java/io/bittiger/ads/activity/TopKAds.java

```diff
@@ -0,0 +1,51 @@
+package io.bittiger.ads.activity;
+
+import io.bittiger.ads.model.Ad;
+
+import java.util.*;
+
+public class TopKAds {
+    private static TopKAds instance = null;
+    protected TopKAds() {
+
+    }
+    public static TopKAds getInstance() {
+        if (instance == null) {
+            instance = new TopKAds();
+        }
+        return instance;
+    }
+    public List<Ad> selectTopKAds(List<Ad> rankedAds, int k) {
+        LinkedList<Ad> selectedAds = new LinkedList<Ad>();
+        if (rankedAds == null || rankedAds.size() == 0) {
+            return selectedAds;
+        }
+        PriorityQueue<Ad> minHeap = new PriorityQueue<Ad>(k, new Comparator<Ad>() {
+            public int compare(Ad o1, Ad o2) {
```

Comment:
Thanks for the comment! I use the expression in case of overflow (although may not happen in our case)